app/src/main/java/com/framgia/soundclound/screen/detailgenre/GenreDetailViewModel.java

```diff
@@ -0,0 +1,85 @@
+package com.framgia.soundclound.screen.detailgenre;
+
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+
+import com.framgia.soundclound.BR;
+import com.framgia.soundclound.data.model.Track;
+import com.framgia.soundclound.data.source.TrackDataSource;
+import com.framgia.soundclound.data.source.TrackRepository;
+import com.framgia.soundclound.data.source.remote.TrackRemoteDataSource;
+import com.framgia.soundclound.util.Constant;
+
+import java.util.List;
+
+/**
+ * Created by Sony on 1/5/2018.
+ */
+
+public class GenreDetailViewModel extends BaseObservable implements TrackClickListener,
+        MoreInfoClickListener {
+
+    private GenreDetailAdapter mGenreDetailAdapter;
+    private String mGenre;
+
+    private static final String TAG = GenreDetailViewModel.class.toString();
```

Comment:
2.8 Class member ordering
This is not the only correct solution, but it is recommended to use the following order :

Constants
Fields
Constructors
Override methods and callbacks (public or private)
Public methods
Private methods
Inner classes or interfaces
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md